clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -66,16 +74,31 @@
 
   private final ServerStoreProxy storeProxy;
   private final Store<K, V> underlyingStore;
+  private final OperationsCodec<K, V> codec;
+  private final ChainResolver<K, V> resolver;
 
-  private ClusteredStore(ServerStoreProxy serverStoreProxy, Store<K, V> underlyingStore) {
+  private ClusteredStore(ServerStoreProxy serverStoreProxy, Store<K, V> underlyingStore,
+                         final OperationsCodec<K, V> codec, final ChainResolver<K, V> resolver) {
     this.storeProxy = serverStoreProxy;
     this.underlyingStore = underlyingStore;
+    this.codec = codec;
+    this.resolver = resolver;
   }
 
   @Override
   public ValueHolder<V> get(final K key) throws StoreAccessException {
-    // TODO: Make appropriate ServerStoreProxy call
-    return underlyingStore.get(key);
+    Chain chain = storeProxy.get(key.hashCode());
+    Map.Entry<Operation<K>, Chain> entry = resolver.resolve(chain, key);
```

Comment:
I had missed this usage of Map.Entry as a tuple. I am pretty sure there is a type hiding there which would make sense ... or an indication that something is not right and needs refactoring.